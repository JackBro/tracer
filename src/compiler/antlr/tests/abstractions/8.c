// example given by Jorge
void main(){
 int x,y,z;
 int a[5];	     

 x=0;
 y=4;	    
 z=5;	        
 _TRACER_abstract(x, y, a[0], a[1], x>0 && y>0 && a[0] >2 && a[0] == a[1]);
}

