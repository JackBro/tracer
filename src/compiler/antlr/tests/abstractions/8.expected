trans(func_main, i6,
	[p$init_0],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[A],[ReturnValue,X,Y,Z],[SA],[SReturnValue,SX,SY,SZ],
	[p$init_1],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[A],[ReturnValue,X,Y,Z],[SA],[SReturnValue,SX,SY,SZ],
	[],
	[],
	[],
	not_branch).
trans(func_main, i7,
	[p$init_1],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[A],[ReturnValue,X,Y,Z],[SA],[SReturnValue,SX,SY,SZ],
	[p0],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[A],[ReturnValue,X,Y,Z],[SA],[SReturnValue,SX,SY,SZ],
	[],
	[call(func___TRACER_INIT,[RV0|_],[SRV0|_],[])],
	[],
	not_branch).
trans(func_main, i1,
	[p0],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[A],[ReturnValue,X,Y,Z],[SA],[SReturnValue,SX,SY,SZ],
	[p8],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[A],[ReturnValue,X$Next$,Y$Next$,Z$Next$],[SA],[SReturnValue,SX$Next$,SY$Next$,SZ$Next$],
	[assign(SX$Next$,0),assign(SY$Next$,4),assign(SZ$Next$,5)],
	[],
	[],
	not_branch) :-
	X$Next$=0,Y$Next$=4,Z$Next$=5.
trans(func_main, i2,
	[p8],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[A],[ReturnValue,X,Y,Z],[SA],[SReturnValue,SX,SY,SZ],
	[p9],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[A],[ReturnValue,X,Y,Z],[SA],[SReturnValue,SX,SY,SZ],
	[],
	[],
	[],
	not_branch).
trans(func_main, i3,
	[p9],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[A],[ReturnValue,X,Y,Z],[SA],[SReturnValue,SX,SY,SZ],
	[p10],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[A],[ReturnValue$Next$,X,Y,Z],[SA],[SReturnValue$Next$,SX,SY,SZ],
	[assign(SReturnValue$Next$,_)],
	[],
	[],
	not_branch) :-
	ReturnValue$Next$=_.
trans(func___TRACER_INIT, i4,
	[p0],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[],[ReturnValue],[],[SReturnValue],
	[p1],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[],[ReturnValue],[],[SReturnValue],
	[],
	[],
	[],
	not_branch).
trans(func___TRACER_INIT, i5,
	[p1],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[],[ReturnValue],[],[SReturnValue],
	[p2],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[],[ReturnValue$Next$],[],[SReturnValue$Next$],
	[assign(SReturnValue$Next$,_)],
	[],
	[],
	not_branch) :-
	ReturnValue$Next$=_.
%--------------------------------------------------
block(b1, func_main, [_], [_], _, nl, [i1,i2,i3,i6,i7],  [n,n],[],[n],[u,u,u,u]).
block(b2, func___TRACER_INIT, [_], [_], _, nl, [i4,i5],  [n,n],[],[],[u]).
%--------------------------------------------------
start(func_main, [p$init_0]).
start(func___TRACER_INIT, [p0]).
%--------------------------------------------------
bottom(func_main, [p10]).
bottom(func___TRACER_INIT, [p2]).
%--------------------------------------------------
freshglobalarrays([SH$1$,SH$2$]).
%--------------------------------------------------
freshglobalvars([]).
%--------------------------------------------------
freshlocalarrays(func_main,[SA]).
freshlocalarrays(func___TRACER_INIT,[]).
%--------------------------------------------------
freshlocalvars(func_main,[SReturnValue,SX,SY,SZ]).
freshlocalvars(func___TRACER_INIT,[SReturnValue]).
%--------------------------------------------------
freshparams(func_main,[lv(0)]).
freshparams(func___TRACER_INIT,[lv(0)]).
%--------------------------------------------------
init_goal(func_main, [p$init_0], i6, [H$1$,H$2$],[],[SH$1$,SH$2$],[],[A],[ReturnValue,X,Y,Z],[SA],[SReturnValue,SX,SY,SZ],[]).
%--------------------------------------------------
abstract(func_main,[p9],
	[H$1$,H$2$],[],[A],[ReturnValue,X,Y,Z],
	[SH$1$,SH$2$],[],[SA],[SReturnValue,SX,SY,SZ],
	[_,_],[],[o],[_,o,o,_],
	[gt(SX,0),gt(SY,0),gt(ref(SA,0),2),eq(ref(SA,0),ref(SA,1))]).
